docs/two-phase-commit-transactions.md

```diff
@@ -500,6 +432,287 @@ public class ServiceBImpl implements ServiceB {
+
+The APIs for mutation operations (`put()`/`delete()`/`mutate()`) could also throw `UnsatisfiedConditionException`.
+If you can this exception, it indicates that the condition for the mutation operation is not met.
+You can handle this exception according to your application requirements.
+
+The `prepare()` API could throw `PreparationException` and `PreparationConflictException`.
+If you catch `PreparationException`, it indicates that preparing the transaction fails due to transient or nontransient faults. You can try retrying the transaction from the beginning, but the transaction may still fail if the cause is nontransient.
+If you catch `PreparationConflictException`, it indicates that preparing the transaction has failed due to transient faults (e.g., a conflict error). You can retry the transaction from the beginning.
+
+The `validate()` API could throw `ValidationException` and `ValidationConflictException`.
+If you catch `ValidationException`, it indicates that validating the transaction fails due to transient or nontransient faults. You can try retrying the transaction from the beginning, but the transaction may still fail if the cause is nontransient.
+If you catch `ValidationConflictException`, it indicates that validating the transaction has failed due to transient faults (e.g., a conflict error). You can retry the transaction from the beginning.
+
+Also, the `commit()` API could throw `CommitException`, `CommitConflictException`, and `UnknownTransactionStatusException`.
+If you catch `CommitException`, it indicates that committing the transaction fails due to transient or nontransient faults. You can try retrying the transaction from the beginning, but the transaction may still fail if the cause is nontransient.
+If you catch `CommitConflictException`, it indicates that committing the transaction has failed due to transient faults (e.g., a conflict error). You can retry the transaction from the beginning.
+If you catch `UnknownTransactionStatusException`, it indicates that the status of the transaction, whether it has succeeded or not, is unknown.
+In such a case, you need to check if the transaction is committed successfully or not and retry it if it fails.
+How to identify a transaction status is delegated to users.
+You may want to create a transaction status table and update it transactionally with other application data so that you can get the status of a transaction from the status table.
+
+Although not illustrated in the sample code, the `resume()` API could also throw a `TransactionNotFoundException`.
+This exception indicates that the transaction associated with the specified ID was not found, and it might have been expired.
+In such cases, you can retry the transaction from the beginning since the cause of this exception is basically transient.
+
+In the sample code, for `UnknownTransactionStatusException`, the transaction doesn't retry because the cause of the exception is nontransient.
```

Comment:
> because the cause of the exception is nontransient

Maybe we should also mention that the transaction status is unknown (due to underlying database issue or something) and whether to retry must be decided by the application?

Reply:
I think that's mentioned here:
https://github.com/scalar-labs/scalardb/pull/934/files#diff-2677f880f1a9e512f31cce8e12950863ddffd4be49408d310d72d4d73e6122b9R676-R679

What do you think?

Reply:
Yeah, the previous description looks perfect.

But, I have a concern that the readers might forget the previous important explanation about UnknownTransactionStatusException since this line only mentions nontrasient causes.

In my understanding, UnknownTransactionStatusException must not be retried, while other non-transient exceptions can be retried although it may make no sense. Just mentioning nontransient causes seems a bit weak in preventing users from attempting retries.

Reply:
That makes sense.

So how about this:
```suggestion
In the sample code, for `UnknownTransactionStatusException`, the transaction doesn't retry as mentioned earlier.
```

Reply:
Looks good!